### DISCUSSION TITLE
How to pass parameters to config.toml.tmpl files

### DISCUSSION THREAD

**Discussion by liyang516** [rancher/rke2#5088] [Q&A]
I refer to https://docs.rke2.io/advanced and https://github.com/k3s-io/k3s/blob/master/pkg/agent/templates/templates_linux.go docs，
I know /var/lib/rancher/rke2/agent/etc/containerd/config.toml.tmpl is the customization file， but I don't know how to pass parameters to config.toml.tmpl files.

For example：
```shell
[plugins."io.containerd.grpc.v1.cri"]
{{- if .DisableCgroup}}
  disable_cgroup = true
{{end}}
``` 

So where or how to set the value of "DisableCgroup" ?

**Comment by brandond** [comment]
You don't. The bound data object is defined here:
https://github.com/k3s-io/k3s/blob/3f237230350b5170eef4e54c7826d88433182efc/pkg/agent/templates/templates.go#L14-L23

The bound data object is filled with values here:
https://github.com/k3s-io/k3s/blob/3f237230350b5170eef4e54c7826d88433182efc/pkg/agent/containerd/config_linux.go#L68-L77

You'll note that most of the values there are determined based on other config, or the state of the running host. This is by design.

**Reply by liyang516** [reply]
got it
